Library/OcAppleKernelLib/KextPatcher.c

```diff
@@ -282,23 +365,28 @@ PatcherBlockKext (
     return EFI_UNSUPPORTED;
   }
 
-  KmodOffset = Context->VirtualKmod - Context->VirtualBase;
-  KmodInfo   = (KMOD_INFO_64_V1 *)((UINT8 *) MachoGetMachHeader64 (&Context->MachContext) + KmodOffset);
-  StartAddr  = KcFixupValue (KmodInfo->StartAddr, NULL);;
+  KmodOffset = Context->VirtualKmod - Context->VirtualBase + Context->FileOffset;
+  KmodInfo   = (UINT8 *) MachoGetMachHeader (&Context->MachContext) + KmodOffset;
```

Comment:
This looks wrong as it needs to be done after the two checks on `KmodOffset` and `TmpOffset`. (It was wrong in the past as well.)